src/components/TodoItem.tsx

```diff
@@ -0,0 +1,189 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+import className from 'classnames';
+import { FormEvent, useState } from 'react';
+import { Todo } from '../types/Todo';
+import * as postService from '../api/todos';
+import { Error } from '../types/Error';
+
+type Props = {
+  todo: Todo,
+  onDeleteTodo: (todoId: number) => void,
+  loadingIds: number[],
+  setTodos: (value: Todo[]) => void,
+  setLoadingIds: React.Dispatch<React.SetStateAction<number[]>>,
+  todos: Todo[]
+};
+
+const USER_ID = 11707;
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  onDeleteTodo,
+  loadingIds,
+  setTodos,
+  setLoadingIds,
+  todos,
+}) => {
+  const [, setErrorMessage] = useState(Error.None);
```

Comment:
why do we need this state, if we do not need the value

Reply:
Using setErrorMessage instead of directly throwing errors with throw provides more flexibility and control over how errors are managed and displayed in app. Also,  It centralizes error handling within the component, making it easier to manage and customize error responses.

Reply:
Still not fixed from the previous review, It's bad practice to write code like that, need to fix it.
You can create a default function and then call it at the right places with a certain value, and using state without using state is not recommended